tests/unit/test_request_class_resolution.py

```diff
@@ -0,0 +1,72 @@
+from typing import Optional
+
+import pytest
+
+from litestar import Controller, HttpMethod, Litestar, Request, Router, get
+from litestar.types import Empty
+
+router_request = type("router_request", (Request,), {})
+controller_request = type("controller_request", (Request,), {})
+app_request = type("app_request", (Request,), {})
+handler_request = type("local_request", (Request,), {})
+
+test_path = "/test"
+
+
+@pytest.mark.parametrize(
+    "layer, expected",
+    [[0, handler_request], [1, controller_request], [2, router_request], [3, app_request], [None, Request]],
+)
+def test_request_class_resolution_of_layers(layer: Optional[int], expected: Request) -> None:
+    class MyController(Controller):
+        path = test_path
+
+        @get(
+            path="/{path_param:str}",
+        )
+        def test_method(self) -> None:
+            pass
+
+    MyController.test_method._resolved_request_class = Empty if layer != 0 else expected  # type: ignore
+    MyController.request_class = None if layer != 1 else expected  # type: ignore
+    router = Router(path="/users", route_handlers=[MyController], request_class=None if layer != 2 else expected)  # type: ignore
+    app = Litestar(route_handlers=[router], request_class=None if layer != 3 else expected)  # type: ignore
+    route_handler, _ = app.routes[0].route_handler_map[HttpMethod.GET]  # type: ignore
+    layer_map = {
+        0: route_handler,
+        1: MyController,
+        2: router,
+        3: app,
+    }
+    component = layer_map.get(layer)  # type: ignore
+    if component:
+        component.request_class = expected
+        assert component.request_class is expected
+    request_class = route_handler.resolve_request_class()
+    assert request_class is expected
+    if component:
+        component.request_class = None
+        assert component.request_class is None
```

Comment:
Is this portion needed? I saw that it's there in the `test_response_class_resolution` as well, but I feel like this is unnecessary.